File: maxhanna.client/src/services/datacontracts/file-entry.ts
import { FileData } from "./file-data";
import { User } from "./user";
import { Comment } from '../datacontracts/comment'
import { Reaction } from "./reaction";

export class FileEntry {
  id: number;
  fileName: string;
  directory: string;
  visibility: string;
  sharedWith: string;
  user: User;
  isFolder: boolean; 
  fileComments: Array<Comment>;
  date: Date;
  fileData?: FileData;
  fileSize: number;
  fileType: string;
  reactions?: Array<Reaction>;
    
  constructor(id: number, fileName: string, directory: string, visibility: string, sharedWith: string,
    user: User, isFolder: boolean, comments: Array<Comment>, date: Date,
    fileSize: number, fileType: string, reactions?: Array<Reaction>, fileData?: FileData) {
    this.id = id;
    this.fileName = fileName;
    this.directory = directory;
    this.visibility = visibility;
    this.sharedWith = sharedWith;
    this.user = user;
    this.isFolder = isFolder; 
    this.fileComments = comments;
    this.date = date;
    this.fileData = fileData;
    this.fileSize = fileSize;
    this.fileType = fileType;
    this.reactions = reactions;
  } 
}

File: maxhanna.Server/Controllers/DataContracts/Files/FileEntry.cs
using maxhanna.Server.Controllers.DataContracts.Topics;
using maxhanna.Server.Controllers.DataContracts.Users;

namespace maxhanna.Server.Controllers.DataContracts.Files
{
	public class FileEntry
	{
		public int Id { get; set; }
		public FileData? FileData { get; set; }
		public string? FileName { get; set; }
		public string? GivenFileName { get; set; }
		public string? Description { get; set; }
		public string? Directory { get; set; }
		public string? Visibility { get; set; }
		public string? SharedWith { get; set; }
		public User? User { get; set; }
		public int LastUpdatedUserId { get; set; }
		public User? LastUpdatedBy { get; set; }
		public bool IsFolder { get; set; }
		public List<FileComment>? FileComments { get; set; }
		public DateTime Date { get; set; }
		public DateTime? LastUpdated { get; set; }
		public string? FileType { get; set; }
		public int FileSize { get; set; }
		public int? Height { get; set; }
		public int? Width { get; set; }
		public int? Duration { get; set; }
		public List<Reaction>? Reactions { get; set; }
		public List<Topic>? Topics { get; set; }
		public DateTime? LastAccess { get; set; }

		public FileEntry() { }
		public FileEntry(int id)
		{
			Id = id;
		}
	}
}
